client/informant/src/lib.rs

```diff
@@ -101,10 +106,23 @@ where
 			future::ready(())
 		});
 
-	future::join(
+	let net_status_provider = interval(Duration::from_millis(5000)).for_each(|()| async {
+		let status = network.status().await;
+		if let Ok(status) = status {
+			let mut network_status_sink = network_status_sink.clone();
+			let _ = network_status_sink.send(status);
```

Comment:
Is it not possible to move the block of code above (lines 90-105) directly here, and remove the `network_status_sink`/`network_status_stream` variables?

Reply:
Yes. I did not do that, because `let (network_status_sink, network_status_stream) = tracing_unbounded("mpsc_network_status")` creates prometheus vector, so it's a breaking change

Reply:
Oh, I think that's totally fine to change.